From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martin Kampas <martin.kampas@jolla.com>
Date: Fri, 26 Mar 2021 12:05:58 +0100
Subject: [PATCH] cmFileAPI: Allow to control the file API path

Allow postprocessing of the reply before the IDE gets to it.
---
 Source/cmFileAPI.cxx | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/Source/cmFileAPI.cxx b/Source/cmFileAPI.cxx
index 8abb5a82f3198f1948d95689c8d26573a2debce2..4551441fe28e8293ca6b1887c1a99eac39089b6e 100644
--- a/Source/cmFileAPI.cxx
+++ b/Source/cmFileAPI.cxx
@@ -29,8 +29,13 @@
 cmFileAPI::cmFileAPI(cmake* cm)
   : CMakeInstance(cm)
 {
+  std::string prefix;
+  cmSystemTools::GetEnv("CMAKE_FILE_API_PREFIX", prefix);
+  if (prefix.empty())
+      prefix = ".cmake";
+
   this->APIv1 =
-    this->CMakeInstance->GetHomeOutputDirectory() + "/.cmake/api/v1";
+    this->CMakeInstance->GetHomeOutputDirectory() + "/" + prefix + "/api/v1";
 
   Json::CharReaderBuilder rbuilder;
   rbuilder["collectComments"] = false;
